src/Tile/SelectableTileGroup.svelte

```diff
@@ -0,0 +1,48 @@
+<script>
+  /**
+   * Specify the selected tile's
+   * @type {any[]}
+   */
+  export let selectedValues = [];
+
+  /** Set to `true` to disable the tile group */
+  export let disabled = false;
+
+  /** Specify the legend text */
+  export let legend = "";
+
+  import { createEventDispatcher, setContext } from "svelte";
+  import { writable } from "svelte/store";
+
+  const dispatch = createEventDispatcher();
+  const _selectedValues = writable(selectedValues);
+
+  function newArray(value, selected) {
+    let a = [...$_selectedValues];
+    const i = a.indexOf(value);
+    if (selected && i === -1) {
+      a.push(value);
+    } else if (!selected && i > -1) {
+      a.splice(i, 1);
+    }
+    return a;
+  }
+
+  setContext("SelectableTileGroup", {
+    selectedValues: _selectedValues,
+    update: ({ selected, value }) =>
+      _selectedValues.set(newArray(value, selected)),
```

Comment:
- Use the store `.update` method
- Use filter/map Array methods

Reply:
I've updated the code to:

```js
  setContext("SelectableTileGroup", {
    selectedValues: _selectedValues,
    update: ({ selected, value }) =>
      _selectedValues.update(v => {
        const copy = [...v];
        const exists = copy.indexOf(value) > -1;
        if (selected && !exists) {
          copy.push(value);
        }
        else if (!selected && exists) {
          copy.splice(i, 1);
        }
        return copy;
      })
  });
```

What would your suggestion be?

Reply:
Something like

```
_selectedValues.update(_ => {
  if (_.includes(value)) {
    return _.filter(i => i !== value);
  } else {
   // since order does not matter
    return [..._, value]
  }
});
```

Reply:
👍 jedi master.

Reply:
Hold on, what if update receives the value but with `selected=false` and it doesn't already exist on the array. Your going to add the item, even if the selected state is false?

Reply:
Yh the deployed code demonstrates that bug. I've redeployed a fix.